### DISCUSSION TITLE
Can you change the button used for panning in a Flickable?

### DISCUSSION THREAD

**Discussion by Dirleye** [slint-ui/slint#6040] [Q&A]
Flickable by default makes use of left-click to pan around the viewport.
Is there a way to change this to work instead with something like right-click instead?

**Comment by Horbin-Magician** [comment]
I'm not sure.  Can you set `interactive` of `Flickable` to `False`, and use other callbacks (such as `moved` of `TouchArea`) to control the `Flickable` (`viewport-x` and `viewport-y`)?

**Comment by ogoffart** [comment]
There is no such feature right now. (implementing manually with a TouchArea is possible but might not be exactly what you need)
What is your usecase?

**Reply by Dirleye** [reply]
I'm using the Flickable as a canvas on which other elements are drawn that the user will need to interact with.

"Node" elements will need to be draggable using left-click. This works currently if the user holds down left click for half a second or longer on a node (as expected from the Flickable documentation) but this will be far too annoying for end users to deal with when moving nodes a lot.

I'll try implementing it manually at some point as these comments have suggested and mark an answer if it works without any problems.
Thank you for your time!
